private/bufpkg/bufstudioagent/buffer_codec.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bufstudioagent
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	connect "github.com/bufbuild/connect-go"
+	"google.golang.org/protobuf/proto"
+)
+
+// bufferCodec is a `connect.Codec` for use with clients of type
+// `connect.Client[bytes.Buffer, bytes.Buffer]` which does not attempt to parse
+// messages but instead allows the application layer to work on the buffers
+// directly. This is useful for creating proxies.
+type bufferCodec struct {
```

Comment:
needs a constructor

Reply:
We changed the style guide to not require constructors. An empty name here is valid for the connect codec interface so there also wouldn't be anything to check.